docs/ceres.rst

```diff
@@ -0,0 +1,102 @@
+
+A Ceres node represents a single time-series metric, and is composed of at least two data files. A slice
+to store all data points, and an arbitrary key-value metadata file. The minimum required metadata a node
+needs is a ``'timeStep'``. This setting is the finest resolution that can be used for writing. A Ceres
+node however can contain and read data with other, less-precise values in its underlying slice data.
+
+Other metadata keys that may be set for compatibility with Graphite may be ``'retentions'`` ,
+``'xFilesFacter'``, and ``'aggregationMethod'``.
+
+A Ceres slice contains the actual data points in a file. The only other information a slice holds is the
+timestamp of the oldest data point stored, and the resolution. Both of which are encoded as part of its
+filename in the format ``timestamp@resolution``.
+
+Data points in Ceres are stored on-disk as a contiguous list of big-endian double-precision floats. The
+timestamp of a datapoint is not stored with the value, rather it is calculated by using the timestamp
+of the slice plus the index offset of the value multiplied by the resolution.
+
+The timestamp is the number of seconds since the UNIX Epoch (01-01-1970). The data value is parsed by the
+Python `float() <http://docs.python.org/library/functions.html#float>`_ function and as such behaves in
+the same way for special strings such as ``'inf'``. Maximum and minimum values are determined by the
+Python interpreter's allowable range for float values which can be found by executing::
+
+    python -c 'import sys; print sys.float_info'
+
+
+Slices: Precision and Fragmentation
+-----------------------------------
+Ceres databases contain one or more slices, each with a specific data resolution and a timestamp to mark
+the start time of the slice. Slices are ordered from the most recent timestamp to the oldest timestamp.
+Resolution of data is not considered when reading data, only that when writing a slice with the highest
+precision configured for the node exists.
+
+Gaps in data are handled in Ceres by padding the data files with null datapoints. If the slice gap
+however is too big, then a new slice is instead created. If a Ceres node accumulates too many slices,
+performance can suffer. This can be caused by intermittently reported data. To mitigate slice
+fragmentation there is a tolerance for how much space can be wasted within a slice file to avoid
+creating a new one. That tolerance level is determined by ``'MAX_SLICE_GAP'``, which is the number of
+consecutive null datapoints allowed in a slice file.
+
+If set very low, Ceres will waste less of the tiny bit disk space that this feature wastes, but then
+will be prone to performance problems caused by slice fragmentation, which can be pretty severe.
+
+If set really high, Ceres will waste a bit more disk space. Although each null datapoint wastes 8 bytes,
+you must keep in mind your filesystem's block size. If you suffer slice fragmentation issues, you should
```

Comment:
Could say something about filesystem compression level (ZFS, Btrfs, Zram, ...).  Then again, maybe not.

Just thinking aloud. :-)